common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -661,11 +663,50 @@ static boolean copy(FileSystem srcFS, Path src,
       // is tried and it fails. We depend upon that behaviour in cases like replication,
       // wherein if distcp fails, there is good reason to not plod along with a trivial
       // implementation, and fail instead.
+      boolean shouldPreserveXAttrs = shouldPreserveXAttrs(conf, srcFS, dstFS);
+      Map<Path, Map<String, byte[]>> XAttrsToDestMapping = null;
+      if (shouldPreserveXAttrs) {
+        XAttrsToDestMapping = new HashMap<>();
+        fetchXAttrs(XAttrsToDestMapping, srcFS, srcFS.getFileStatus(src), dst);
+      }
       copied = FileUtil.copy(srcFS, src, dstFS, dst, deleteSource, overwrite, conf);
+      if (copied && shouldPreserveXAttrs) {
+        for (Map.Entry<Path, Map<String, byte[]>> xAttrs : XAttrsToDestMapping.entrySet()) {
+          for (Map.Entry<String, byte[]> val : xAttrs.getValue().entrySet()) {
+            dstFS.setXAttr(xAttrs.getKey(), val.getKey(), val.getValue());
+          }
+        }
+      }
     }
     return copied;
   }
 
+  public static void fetchXAttrs(Map<Path, Map<String, byte[]>> xAttrs, FileSystem srcFS, FileStatus srcStatus, Path dst) throws IOException {
+    Path src = srcStatus.getPath();
+    Path dstFile = new Path(dst, src.getName());
```

Comment:
Check the path resolution, It won't work in couple of cases like if src is a file & /dst doesn't exist.
Can extract the logic for FileUtils.copy from hadoop & have a preserveXAttr flag additional